src/near_web3_extensions.js

```diff
@@ -0,0 +1,56 @@
+const utils = require('./utils')
+
+module.exports = function (web3) {
+	/**
+	 * Formats the input of retrieveNear and converts all values to HEX
+	 *
+	 * @method inputRetrieveNearFormatter
+	 * @param {Object} options
+	 * @returns object
+	 */
+	var inputRetrieveNearFormatter = function (options) {
+		// TODO: check that valid near accountId
+
+		// allow both
+		if (options.gas || options.gasLimit) {
+			options.gas = options.gas || options.gasLimit;
+		}
+
+	    options = _convertOptionsToHex(options);
+
+	    // check from, only if not number, or object
+	    if (typeof(options.from) !== 'number' && typeof(options.from) !== 'object') {
+	        options.from = options.from || (this ? this.defaultAccount : null);
```

Comment:
Is there a time where `this` would be `undefined`?

Reply:
honestly, this is a hacky copy-paste. I took their transaction formatter and tweaked some thing for retrieve near. I may be table to just take this out. I'll play around with it

Reply:
I wish their transactions took options as separate args instead of an obj, I think it would be cleaner bc then I'd just need to change the `to`

Reply:
looks like this `this` is acting like it should be, and is never undefined. I'm afraid to take out this conditional, just bc they had it...lol.

Reply:
> honestly, this is a hacky copy-paste. I took their transaction formatter and tweaked some thing for retrieve near. I may be table to just take this out. I'll play around with it

My favorite kind of paste.

Reply:
> I wish their transactions took options as separate args instead of an obj, I think it would be cleaner bc then I'd just need to change the `to` formatter. Tweaking these formatters, it seems to me, will require enough copy-paste to be uncomfortable. Any _huge_ architectural changes to web3 would be a headache to keep up with.

Yeah definitely.  Hopefully this isn't something that would change in any web3 overhauls, but just in case, can you add a comment about it in the code to not touch this without verifying compatibility with web3?

Reply:
Hmm...so everything with this should stay the same _except_ that we need to do something different with the `to` field?

What if everything is passed through the `inputTransactionFormatter` first and then the `to` field is modified? If it doesn't like the `to` field we currently pass in, you could pass through a dummy value to get it to pass so everything else gets modified as needed, then do what you need to do with the `to` field.

Reply:
excellent hacking skillz. I like it. changing now.